src/pydartdiags/stats/stats.py

```diff
@@ -214,6 +225,37 @@ def bin_by_layer(df, levels, verticalUnit="pressure (Pa)"):
     df.loc[:, "midpoint"] = df["vlevels"].apply(lambda x: x.mid)
 
 
+def bin_by_time(df, time_value):
+    """
+    Bin observations by time and add 'time_bin' and 'time_bin_midpoint' columns to the DataFrame.
+
+    Args:
+        df (pd.DataFrame): The input DataFrame containing a 'time' column.
+        time_value (str): The width of each time bin (e.g., '3600S' for 1 hour).
+
+    Returns:
+        None: The function modifies the DataFrame in place by adding 'time_bin' and 'time_bin_midpoint' columns.
+    """
+    # Create time bins
+    start = df["time"].min() - timedelta(seconds=1)
+    end = df["time"].max()
+    # Adjust the end time to align with the frequency
+    aligned_end = (pd.Timestamp(end) + pd.Timedelta(time_value)).floor(time_value)
+
```

Comment:
ceiling to capture values exactly on the end?

Reply:
nope bin is (a,b]  which is a < value <=b

`aligned_end = (pd.Timestamp(end) + pd.Timedelta(time_value)).floor(time_value)` get us a bin that covers the max value